### PR TITLE
Qubic Metrics

### PR DESCRIPTION
**Pull request by tomaspozo**
The Qubic Metrics project adds new visualizations that display historic data from different sources in the Qubic ecosystem. These new visualizations lived in the Analytics page, and include the following features:

- Market cap and price values on a daily basis 
- Weekly scores statistics chart
- Solutions tracking chart with a 5 minute granularity.
- Solutions per hour tracking with a 20 minute granularity.
-Several statistics around the Qubic open source repositories like starts, issues, and others.

New libraries:
- For charting we added tremor and its dependencies. All tremor components live under `src/components/tremor`.

Required env variables:
- VITE_METRICS_API_URL: the url of the QubicMetrics backend
- VITE_METRICS_API_TOKEN: a public key to authenticate the API calls

### CODE REVIEW

File: src/pages/analytics/components/ReposHistoryChart.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  if (isLoading) {
+    return <LinearProgress />
+  }
+
+  return (
+    <AreaChart
+      className="h-320"
+      data={statsData || []}
+      index="date"
+      categories={['starsCount', 'commits', 'openIssues', 'closedIssues', 'watchersCount']}
```
Comment (alexmf91) [inline src/pages/analytics/components/ReposHistoryChart.tsx:21]: Put this array data on top of the component, is going to be recreated on every render like it is. like const CATEGORIES = [...]

File: src/pages/analytics/components/ReposHistoryChart.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      colors={['gray', 'gray', 'gray', 'gray', 'gray']}
```
Comment (alexmf91) [inline src/pages/analytics/components/ReposHistoryChart.tsx:22]: Same here

File: src/pages/analytics/components/ScoresHistoryCharts.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+  return (
+    <ChartContainer isLoading={isLoading}>
+      <CardItem className="w-full px-24 py-20">
+        <div className="flex flex-col gap-20">
+          <div className="flex flex-col justify-between gap-20 sm:flex-row sm:gap-8 md:gap-10 lg:gap-20">
+            <div className="flex items-center justify-between gap-8 sm:justify-start">
+              <p className="font-space text-22 font-500">{t('weeklyScores')}</p>
+              <p className="align-middle font-space text-14 text-gray-50">
+                ({calculateVariance(data, 'averageScore')})
```
Comment (alexmf91) [inline src/pages/analytics/components/ScoresHistoryCharts.tsx:38]: wrap this in a useMemo

File: src/pages/analytics/components/SolutionsChart.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import { useQueryState } from 'nuqs'
+
+import { useTranslation } from 'react-i18next'
+
+import { useGetQubicLIScoresHistoryQuery } from '@app/store/apis/metrics-v1.api'
```
Comment (alexmf91) [inline src/pages/analytics/components/SolutionsChart.tsx:5]: group imports by: external libraries, @app.. , and relative imports ( like scope imports)

File: src/pages/analytics/components/SolutionsPerHourChart.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            <div className="flex items-center justify-between gap-8 sm:justify-start">
+              <p className="font-space text-22 font-500">{t('solutionsPerHour')}</p>
+            </div>
+          </div>
+          <AreaChart
+            showLegend={false}
+            className="h-320"
+            data={data || []}
+            index="date"
+            categories={['solutionsPerHour', 'allTimeSolutionsPerHour']}
```
Comment (alexmf91) [inline src/pages/analytics/components/SolutionsPerHourChart.tsx:33]: Put this oustide the component, on top of the file

File: src/pages/analytics/components/SubCardItem.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import { clsxTwMerge } from '@app/utils'
```
Comment (alexmf91) [inline src/pages/analytics/components/SubCardItem.tsx:1]: This component is exatly the same that we have on network, don't duplicate components please

File: src/router/router.tsx [hunk 7]
```diff
@@ -2,6 +2,7 @@ import { createBrowserRouter, Navigate } from 'react-router-dom'
 
 import { AppLayout } from '@app/components/ui/layouts'
 import { Error404Page } from '@app/pages'
+import AnalyticsPage from '@app/pages/analytics/AnalyticsPage'
```
Comment (alexmf91) [inline src/router/router.tsx:5]: Relocate this page under network folder and Lazy load this page following the pattern we use

File: src/store/apis/metrics-v1.api.ts [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+  GithubStatsHistory,
+  GithubStatsOverview,
+  QubicLIScoresStats,
+  QubicLIStats,
+  QubicStats
+} from './metrics-v1.types'
+
+const BASE_URL = `${envConfig.METRICS_API_URL}/stats`
+
+export const metricsV1Api = createApi({
```
Comment (alexmf91) [inline src/store/apis/metrics-v1.api.ts:13]: if there is no live v2 version, just name it as metricsApi, rename it in all places please

File: src/components/tremor/ComboChart.tsx [hunk 9]
```diff
@@ -0,0 +1,1087 @@
+import { useOnWindowResize } from '@app/hooks/useOnWindowResize'
```
Comment (alexmf91) [inline src/components/tremor/ComboChart.tsx:1]: Same here, split this in to smaller files. Don't use any external libraries like remix icons, you can use what we already have and if it's not enough you can add it on assets and folow the pattern we do

File: package.json [hunk 10]
```diff
@@ -21,7 +21,9 @@
   "dependencies": {
     "@headlessui/react": "^2.1.2",
     "@qubic-lib/qubic-ts-library": "0.1.4",
+    "@radix-ui/react-tooltip": "^1.1.3",
```
Comment (alexmf91) [inline package.json:24]: Remove this library. we  already have tooltip component

File: package.json [hunk 11]
```diff
@@ -21,7 +21,9 @@
...
     "@reduxjs/toolkit": "^2.2.6",
+    "@remixicon/react": "^4.2.0",
```
Comment (alexmf91) [inline package.json:26]: Remove this library, use existing icons or add them manually from heroicons

File: src/components/tremor/AreaChart.tsx [hunk 12]
```diff
@@ -0,0 +1,920 @@
+import { RiArrowLeftSLine, RiArrowRightSLine } from '@remixicon/react'
```
Comment (alexmf91) [inline src/components/tremor/AreaChart.tsx:1]: This large files are not acceptable, split this in to small components, also don't use react namespace and external libraries for icons. Components should be max 150 - 300 lines of code if it's possible

**Review by alexmf91 — CHANGES_REQUESTED** [review]
Overall looks good, you did a good job! But there are some things that are required to change to follow project patterns. Also I have seen that ui is not fully responsive. Please take a look to all comments carefully and fix them. Thanks!!!

Screenshot 2024-11-21 at 18 20 45

File: CHANGELOG.md [hunk 13]
```diff
@@ -202,152 +202,152 @@
...
-- Implemented archiver v2 api
-- Migrated archiver trasnactions v1 endpoints to v2
-- Implemented timestamp display on transaction items
-- Updated dev proxies implementation
-- Updated formatDate function to handle splitting dates
-- Updated AddressPage to get latest transactions from archiver v2 api
-- Optimised InfiniteScroll component
-
-
+## What's Changed
```
[the remark this answers is not in the document]
Comment (tomaspozo) [reply CHANGELOG.md]: yeah, looks like an error on commits, I removed that first commit

**Comment by tomaspozo** [comment]
Qubic team has asked that Qubic Metrics be a independent project and code. Closing this PR since we won't be merging this code. Thank you!

**Pull request closed without merging**